### PR TITLE
Make disabling accordion items possible

### LINKED ISSUE #2438: [Enhancement] Disable accordion item
It would be great if it was possible to 'disable' an accordion-item. By disable I mean make it unclickable and possibly change the text.
image

### PR DESCRIPTION
**Pull request by rasmusCCbankdata**
## Which issue does this PR close?
This PR closes #2438

## What is the new behavior?
Possible to 'disable' an accordion-item. 

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

## Are there any additional context?
See issue https://github.com/kirbydesign/designsystem/issues/2438

#####
I'm a Kirby and Angular Animations noob, so please review carefully and consider this a draft.

### CODE REVIEW

File: apps/cookbook/src/app/examples/accordion-example/accordion-example.component.html [hunk 1]
```diff
@@ -8,3 +8,6 @@ <h2>Accordion with item initially expanded</h2>
 
 <h2>Accordion inside a card</h2>
 <cookbook-accordion-card-example></cookbook-accordion-card-example>
+
+<h2>Accordion with disabled item</h2>
+<cookbook-accordion-disabled-example></cookbook-accordion-disabled-example>
```
Comment (RasmusKjeldgaard) [inline apps/cookbook/src/app/examples/accordion-example/accordion-example.component.html:13]: I actually think we could just make e.g. Accordion Item 3 in the default example disabled instead. Dont think we need a new example for this.

**Review by RasmusKjeldgaard — CHANGES_REQUESTED** [review]
Thank you for an awesome contribution. I have a couple of suggestions for the documentation, and there is also a tiny bit of design refinement to be done by @henrikvoetmand, that we should include when he has published the design he is working on to zeplin.

File: libs/designsystem/src/lib/components/accordion/accordion-item.component.scss [hunk 2]
```diff
@@ -35,6 +35,15 @@ $padding: utils.size('s');
   padding: 0 $padding $padding $padding;
 }
 
+.disabled {
+  cursor: default;
+
+  .title {
+    color: utils.get-text-color('semi-dark');
+    font-weight: utils.font-weight('light');
```
[the remark this answers is not in the document]
Comment (henrikvoetmand) [reply libs/designsystem/src/lib/components/accordion/accordion-item.component.scss:43]: The text should be "semi-dark text" (#707070) and icon shluld be "semi-dark" (#8E8E8E)

File: libs/designsystem/src/lib/components/accordion/accordion-item.component.ts [hunk 3]
```diff
@@ -17,9 +17,29 @@ let uniqueId = 0;
...
+    if (this.isDisabled && !!this.disabledTitle) {
+      return this.disabledTitle;
+    } else {
+      return this.title;
+    }
+  }
+
+  toggleExpanded() {
+    this.isExpanded = !this.isExpanded && !this.isDisabled;
+  }
```
Comment (jkaltoft) [inline libs/designsystem/src/lib/components/accordion/accordion-item.component.ts:42]: See other comment about Kirby v7 conflicts

**Review by jkaltoft — COMMENTED** [review]
Nice contribution. Thank you!

Depending on when this enhancement is scheduled for release we may need to rebase the branch on the Kirby v7 release branch.

File: libs/designsystem/src/lib/components/accordion/accordion-item.component.html [hunk 4]
```diff
@@ -1,14 +1,15 @@
 <ng-container>
   <div
-    (click)="isExpanded = !isExpanded"
+    (click)="toggleExpanded()"
```
[the remark this answers is not in the document]
Comment (RasmusKjeldgaard) [reply libs/designsystem/src/lib/components/accordion/accordion-item.component.html:3]: @jkaltoft this was the one I talked about during standup. Would probably be good to release friday as part of our ordinary release flow, as the team awaits the changes.

[the remark this answers is not in the document]
Comment (rasmusCCbankdata) [reply libs/designsystem/src/lib/components/accordion/accordion-item.component.html:3]: I have changed the name to _onToggleExpanded

**Review by RasmusKjeldgaard — APPROVED** [review]
With the new changes I think we are there. 

@jkaltoft are you OK with this as is? Then we will just have to do a bit of merging to make sure it works with v7, which I think is ok.

**Review by RasmusKjeldgaard — APPROVED** [review]

**Pull request merged**